dalm/datasets/reading_comprehension_generation/README.md

```diff
@@ -0,0 +1,50 @@
+## A note about reading comprehension
+
+This aproach of adapting LLMs is based on this [paper](https://arxiv.org/abs/2309.09530) by Microsoft 
+The way espoused by the paper is generating reading comprehension questions and answers based on the raw corpora
+and training a llm on said generated dataset can enhance its domain adaptiveness
+
+We have two ways of generating  reading comprehension data
+
+1. Via regex based methods that combs the input data for match and aligns them into questions and answers
+2. Via prompting a large language model to come up with questions and answers 
```

Comment:
As a user, how should I decide which approach to use?  Here's a stab:

* Use regex based reading comprehension dataset generation when it works on that dataset
* Otherwise fallback to the slower synthetic data generation approach